util/test/demos/apple/apple_window.cpp

```diff
@@ -0,0 +1,192 @@
+/******************************************************************************
+* The MIT License (MIT)
+*
+* Copyright (c) 2022 Baldur Karlsson
+*
+* Permission is hereby granted, free of charge, to any person obtaining a copy
+* of this software and associated documentation files (the "Software"), to deal
+* in the Software without restriction, including without limitation the rights
+* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+* copies of the Software, and to permit persons to whom the Software is
+* furnished to do so, subject to the following conditions:
+*
+* The above copyright notice and this permission notice shall be included in
+* all copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+* THE SOFTWARE.
+******************************************************************************/
+
+#include "apple_window.h"
+
+#define NS_PRIVATE_IMPLEMENTATION
+#define CA_PRIVATE_IMPLEMENTATION
+#include "official/metal-cpp.h"
+
+static NS::Application *pSharedApplication = NULL;
+MyAppDelegate *AppleWindow::pAppDelegate = NULL;
+
+class MyAppDelegate : public NS::ApplicationDelegate
+{
+public:
+  ~MyAppDelegate();
+
+  NS::Menu *createMenuBar();
+
+  virtual void applicationWillFinishLaunching(NS::Notification *pNotification) override;
+  virtual void applicationDidFinishLaunching(NS::Notification *pNotification) override;
+  virtual bool applicationShouldTerminateAfterLastWindowClosed(NS::Application *pSender) override;
+
+  void CreateWindow(int width, int height, const char *title);
+  NS::View *GetContentView();
+  NS::Window *GetWindow();
+
+private:
+  NS::Window *_pWindow;
+};
+
+MyAppDelegate::~MyAppDelegate()
+{
+  _pWindow->release();
+}
+
+NS::Menu *MyAppDelegate::createMenuBar()
+{
+  using NS::StringEncoding::UTF8StringEncoding;
+
+  NS::Menu *pMainMenu = NS::Menu::alloc()->init();
+  NS::MenuItem *pAppMenuItem = NS::MenuItem::alloc()->init();
+  NS::Menu *pAppMenu = NS::Menu::alloc()->init(NS::String::string("Appname", UTF8StringEncoding));
+
+  NS::String *appName = NS::RunningApplication::currentApplication()->localizedName();
+  NS::String *quitItemName =
+      NS::String::string("Quit ", UTF8StringEncoding)->stringByAppendingString(appName);
+  SEL quitCb =
+      NS::MenuItem::registerActionCallback("appQuit", [](void *, SEL, const NS::Object *pSender) {
+        auto pApp = NS::Application::sharedApplication();
```

Comment:
I don't strictly require matching codestyle in the tests project but it's definitely preferred to follow the general renderdoc guidelines - avoiding use of auto in places like this in favour of the concrete type.